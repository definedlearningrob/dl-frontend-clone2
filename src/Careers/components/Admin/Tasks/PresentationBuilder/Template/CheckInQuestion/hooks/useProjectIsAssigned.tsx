import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
