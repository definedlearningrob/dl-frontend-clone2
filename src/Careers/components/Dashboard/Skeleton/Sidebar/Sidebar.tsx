import { DCLogo } from '@dc/shared/DCLogo/DCLogo';

import SkeletonSidebarFooter from './Footer/Footer';
import SkeletonSidebarItem from './Item/Item';

type Props = {
  itemsCount: number;
};

function SkeletonSidebar({ itemsCount }: Props) {
  const renderItems = () =>
    Array.from({ length: itemsCount }, (_, index) => <SkeletonSidebarItem key={index} />);

  return (
    <div className='skeleton-sidebar'>
      <DCLogo className='skeleton-sidebar__logo-icon' />
      {renderItems()}
      <div className='skeleton-sidebar__divider' />
      <SkeletonSidebarFooter />
    </div>
  );
}

export default SkeletonSidebar;
