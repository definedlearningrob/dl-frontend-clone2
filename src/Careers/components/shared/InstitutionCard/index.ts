export * from './InstitutionCard';
export * from './InstitutionCardThumbnail';
