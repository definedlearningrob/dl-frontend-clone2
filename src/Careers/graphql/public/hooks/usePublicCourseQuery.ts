import { useQuery } from '@apollo/client';

import { PUBLIC_COURSE_QUERY } from '@dc/graphql/public/queries/course';

type Params = {
  skip?: boolean;
  code: string;
  shareId: string;
};

export const usePublicCourseQuery = ({ skip, code, shareId }: Params) =>
  useQuery(PUBLIC_COURSE_QUERY, {
    skip,
    variables: { code, shareId, track: true },
  });
