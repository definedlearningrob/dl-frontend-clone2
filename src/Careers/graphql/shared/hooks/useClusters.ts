import { useQuery } from '@apollo/client';

import CLUSTERS, { TClustersData } from '../queries/clusters';

export const useClusters = () => useQuery<TClustersData>(CLUSTERS);
