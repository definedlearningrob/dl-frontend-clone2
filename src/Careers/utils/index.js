const isObject = (object) => object !== null && typeof object === 'object';

export const isEqual = (object1, object2) => {
  if (object1 === object2) return true;

  if (
    object1 === null ||
    object2 === null ||
    typeof object1 !== 'object' ||
    typeof object2 !== 'object'
  )
    return false;

  const keys1 = Object.keys(object1);
  const keys2 = Object.keys(object2);

  if (keys1.length !== keys2.length) {
    return false;
  }

  for (let key of keys1) {
    const val1 = object1[key];
    const val2 = object2[key];
    const areObjects = isObject(val1) && isObject(val2);
    if (
      // eslint-disable-next-line
      areObjects && !isEqual(val1, val2) ||
      // eslint-disable-next-line
      !areObjects && val1 !== val2
    ) {
      return false;
    }
  }

  return true;
};

export const getByStringKey = (object, key) => {
  const keys = key.split('.');

  return keys.reduce((acc = {}, key) => acc[key], object);
};
