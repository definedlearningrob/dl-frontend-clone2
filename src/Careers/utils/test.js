/* eslint-disable react/prop-types */
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { Router, Route } from 'react-router-dom';

import i18n from '@dc/i18n';

export function renderWithRouter(
  ui,
  { route = '/', routePath = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>
      <I18nextProvider i18n={i18n}>
        <Route path={routePath} render={() => children} />
      </I18nextProvider>
    </Router>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
    history,
  };
}

export function renderWithReduxProvider(
  ui,
  {
    initialState = {
      session: {
        loginError: {},
        user: null,
      },
    },
    store = configureMockStore([thunk])(initialState),
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </Provider>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}

export function renderWithRouterAndReduxProvider(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {
      session: {
        loginError: {},
        user: {},
      },
    },
    store = configureMockStore([thunk])(initialState),
  } = {},
  ...options
) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <Router history={history}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </Router>
    </Provider>
  );

  return {
    ...render(ui, { wrapper: Wrapper, ...options }),
    history,
  };
}

export function renderWithI18N(ui) {
  const Wrapper = ({ children }) => <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}
