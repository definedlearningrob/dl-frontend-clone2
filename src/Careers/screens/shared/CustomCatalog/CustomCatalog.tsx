import { useCustomCatalogQuery } from '@graphql/dc/shared/hooks';
import { useEffect } from 'react';

import { Catalog } from '@shared/components/Catalog';
import { MainContent } from '@shared/components/MainContent/MainContent';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';

export const CustomCatalog = () => {
  const { setBackNavButton } = useNavigation();
  const { data, loading } = useCustomCatalogQuery();

  useEffect(() => {
    setBackNavButton(true);

    return () => {
      setBackNavButton(false);
    };
  }, []);

  return (
    <MainContent>
      <Catalog catalog={data?.careersCatalog} isLoading={loading} />
    </MainContent>
  );
};
