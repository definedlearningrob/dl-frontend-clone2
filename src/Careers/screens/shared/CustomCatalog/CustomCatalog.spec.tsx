import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { CustomCatalogQuery } from '@graphql/dc/shared/operations';
import { CustomCatalogDocument } from '@graphql/dc/shared/hooks';
import { screen, within } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';

import { renderWithRouter } from '@dc/utils/test';

import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';

import { CustomCatalog } from './CustomCatalog';

const catalogMock: MockedResponse<CustomCatalogQuery> = {
  request: { query: CustomCatalogDocument },
  result: {
    data: {
      careersCatalog: {
        id: '1',
        name: 'Custom catalog',
        description:
          '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ut diam ut orci pretium rhoncus at quis justo. Integer laoreet tellus ac tincidunt viverra. Phasellus hendrerit pulvinar mi, vel blandit eros. In at dolor vulputate, ullamcorper urna a, pharetra leo. Vivamus hendrerit pellentesque ullamcorper. Mauris ligula turpis, posuere sit amet euismod quis, vulputate ut nisi. Nam laoreet maximus ex ac pellentesque. Quisque pharetra a dolor non vestibulum. Nulla luctus quam eget massa mollis feugiat. Morbi tincidunt odio suscipit sapien tincidunt commodo.</p>',
        imageUrl: 'https://via.placeholder.com/150',
        thumbnailUrl: 'https://via.placeholder.com/150',
        tracks: [
          {
            id: '11',
            grades: ['K', '1'],
            name: 'Track 1',
            resourcesCount: 10,
            shortDescription: 'Short description',
            imageUrl: 'https://via.placeholder.com/150',
            thumbnailUrl: 'https://via.placeholder.com/150',
          },
          {
            id: '12',
            grades: ['1', '2', '3'],
            name: 'Track 2',
            resourcesCount: 2,
            shortDescription: 'Short description 2',
            imageUrl: 'https://via.placeholder.com/150',
            thumbnailUrl: 'https://via.placeholder.com/150',
          },
          {
            id: '13',
            grades: ['7'],
            name: 'Track 3',
            resourcesCount: 8,
            shortDescription: 'Short description 3',
            imageUrl: 'https://via.placeholder.com/150',
            thumbnailUrl: 'https://via.placeholder.com/150',
          },
        ],
      },
    },
  },
};

const renderComponent = () =>
  renderWithRouter(
    <div app-type='careers' className='app'>
      <MockedProvider mocks={[catalogMock]}>
        <NavigationContextProvider>
          <CustomCatalog />
        </NavigationContextProvider>
      </MockedProvider>
    </div>
  );

describe('CustomCatalog', () => {
  it('renders correctly', async () => {
    const { container } = renderComponent();

    const catalogName = await screen.findByRole('heading', { name: 'Custom catalog' });
    expect(catalogName).toBeInTheDocument();

    const trackCards = screen.getAllByRole('listitem');
    expect(trackCards).toHaveLength(3);

    expect(trackCards[0]).toHaveTextContent('Track 1');
    expect(trackCards[1]).toHaveTextContent('Track 2');
    expect(trackCards[2]).toHaveTextContent('Track 3');

    expect(container).toMatchSnapshot();
  });

  it('navigates to a track after clicking on the card', async () => {
    const { history } = renderComponent();
    history.push = jest.fn();

    const trackCard = await screen.findByRole('listitem', { name: 'Track 1' });
    userEvent.click(within(trackCard).getByRole('link'));

    expect(history.push).toHaveBeenCalledTimes(1);
    expect(history.push).toHaveBeenCalledWith('/catalog/unit-outline/11');
  });
});
