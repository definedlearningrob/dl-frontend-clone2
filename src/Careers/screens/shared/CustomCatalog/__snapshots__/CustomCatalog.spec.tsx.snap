// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCatalog renders correctly 1`] = `
<div>
  <div
    app-type="careers"
    class="app"
  >
    <div
      class="p-base xxxl:p-md"
      id="read"
    >
      <div
        class="flex flex-col gap-base"
      >
        <div
          class="card flex gap-base xxxl:gap-md"
        >
          <img
            class="rounded-sm h-[180px] xxxl:h-[240px] basis-[320px] xxxl:basis-[426px] shrink-0 grow-0 object-cover"
            src="https://via.placeholder.com/150"
          />
          <div
            class="flex flex-col gap-sm xxxl:gap-base"
          >
            <h4
              class="text-base xxxl:text-lg mb-0"
            >
              Custom catalog
            </h4>
            <div
              class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary line-clamp-6"
            >
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ut diam ut orci pretium rhoncus at quis justo. Integer laoreet tellus ac tincidunt viverra. Phasellus hendrerit pulvinar mi, vel blandit eros. In at dolor vulputate, ullamcorper urna a, pharetra leo. Vivamus hendrerit pellentesque ullamcorper. Mauris ligula turpis, posuere sit amet euismod quis, vulputate ut nisi. Nam laoreet maximus ex ac pellentesque. Quisque pharetra a dolor non vestibulum. Nulla luctus quam eget massa mollis feugiat. Morbi tincidunt odio suscipit sapien tincidunt commodo.
              </p>
            </div>
          </div>
        </div>
        <ul
          class="flex gap-sm"
        >
          <li
            aria-label="Track 1"
            class="basis-[320px] xxxl:basis-[520px]"
          >
            <a
              class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
              href="/catalog/unit-outline/11"
            >
              <div
                class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
              >
                <img
                  class="rounded-t-sm h-[112px] xxxl:h-[140px] w-full object-cover"
                  src="https://via.placeholder.com/150"
                />
              </div>
              <div
                class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
              >
                <div
                  class="flex gap-xxs xxxl:gap-xs mb-xs whitespace-nowrap"
                >
                  <span
                    class="min-w-0"
                    data-state="closed"
                  >
                    <div
                      class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                    >
                      <div
                        class="truncate"
                      >
                        Grades: K - 1
                      </div>
                    </div>
                  </span>
                  <span
                    class="truncate"
                    data-state="closed"
                  >
                    <div
                      class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                    >
                      <div
                        class="truncate"
                      >
                        Resources: 10
                      </div>
                    </div>
                  </span>
                </div>
                <span
                  data-state="closed"
                >
                  <h6
                    class="text-xs xxxl:text-sm line-clamp-2"
                  >
                    Track 1
                  </h6>
                </span>
                <div
                  class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6 mb-0"
                >
                  Short description
                </div>
              </div>
              <div
                class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
              >
                <span
                  class="text-xs xxxl:text-sm leading-lg font-medium"
                >
                  View
                </span>
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
              </div>
            </a>
          </li>
          <li
            aria-label="Track 2"
            class="basis-[320px] xxxl:basis-[520px]"
          >
            <a
              class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
              href="/catalog/unit-outline/12"
            >
              <div
                class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
              >
                <img
                  class="rounded-t-sm h-[112px] xxxl:h-[140px] w-full object-cover"
                  src="https://via.placeholder.com/150"
                />
              </div>
              <div
                class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
              >
                <div
                  class="flex gap-xxs xxxl:gap-xs mb-xs whitespace-nowrap"
                >
                  <span
                    class="min-w-0"
                    data-state="closed"
                  >
                    <div
                      class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                    >
                      <div
                        class="truncate"
                      >
                        Grades: 1 - 3
                      </div>
                    </div>
                  </span>
                  <span
                    class="truncate"
                    data-state="closed"
                  >
                    <div
                      class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                    >
                      <div
                        class="truncate"
                      >
                        Resources: 2
                      </div>
                    </div>
                  </span>
                </div>
                <span
                  data-state="closed"
                >
                  <h6
                    class="text-xs xxxl:text-sm line-clamp-2"
                  >
                    Track 2
                  </h6>
                </span>
                <div
                  class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6 mb-0"
                >
                  Short description 2
                </div>
              </div>
              <div
                class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
              >
                <span
                  class="text-xs xxxl:text-sm leading-lg font-medium"
                >
                  View
                </span>
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
              </div>
            </a>
          </li>
          <li
            aria-label="Track 3"
            class="basis-[320px] xxxl:basis-[520px]"
          >
            <a
              class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
              href="/catalog/unit-outline/13"
            >
              <div
                class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
              >
                <img
                  class="rounded-t-sm h-[112px] xxxl:h-[140px] w-full object-cover"
                  src="https://via.placeholder.com/150"
                />
              </div>
              <div
                class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
              >
                <div
                  class="flex gap-xxs xxxl:gap-xs mb-xs whitespace-nowrap"
                >
                  <span
                    class="min-w-0"
                    data-state="closed"
                  >
                    <div
                      class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                    >
                      <div
                        class="truncate"
                      >
                        Grades: 7
                      </div>
                    </div>
                  </span>
                  <span
                    class="truncate"
                    data-state="closed"
                  >
                    <div
                      class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                    >
                      <div
                        class="truncate"
                      >
                        Resources: 8
                      </div>
                    </div>
                  </span>
                </div>
                <span
                  data-state="closed"
                >
                  <h6
                    class="text-xs xxxl:text-sm line-clamp-2"
                  >
                    Track 3
                  </h6>
                </span>
                <div
                  class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6 mb-0"
                >
                  Short description 3
                </div>
              </div>
              <div
                class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
              >
                <span
                  class="text-xs xxxl:text-sm leading-lg font-medium"
                >
                  View
                </span>
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
